src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -358,11 +358,13 @@ private async Task<ActionResult<SynchronizePullRequestResult>> SynchronizePullRe
                 // merge the PR if they are successul.
                 case PrStatus.Open:
                     ActionResult<MergePolicyCheckResult> checkPolicyResult = await CheckMergePolicyAsync(prUrl, darc);
+                    pr.MergePolicyResult = checkPolicyResult.Result;
 
                     switch (checkPolicyResult.Result)
                     {
                         case MergePolicyCheckResult.Merged:
                             await UpdateSubscriptionsForMergedPRAsync(pr.ContainedSubscriptions);
+                            await AddDependencyFlowEventsAsync(pr.ContainedSubscriptions, DependencyFlowEventType.Completed, DependencyFlowEventReason.AutomaticallyMerged, checkPolicyResult.Result);
```

Comment:
Is it going to be a problem if this dies here? If we add the event and then the service exits? Or if the line before runs and this one doesn't?  Are the readers going to be tolerant to having events that don't match the current state, or state changes that don't have the events to go with them?

They are going to have to be, otherwise we'll need some sort of idempotent operation behavior to handle this (to ensure they always match and either both are inserted or neither is).

Reply:
I don't think there will be a problem if we die here. This work is going to be in conjunction with work that @meganaquinn is doing to migrate this data to Kusto, which our plan is to migrate once a day, so the data that we use for the KPIs will always be slightly out of sync with the actual current state. I also think that if we are missing some events, that will be acceptable, but that is something we'll want to bring to our customers to confirm that's ok.

@chcosta do you have any thoughts?

Reply:
Sweet. If it's just KPI data, the occasional dropped (or duplicated) entry shouldn't be a problem.

Reply:
> Are the readers going to be tolerant to having events that don't match the current state

yes

> or state changes that don't have the events to go with them

in some cases yes, but in general no. It's ok for the data to be out of sync, but it would be bad if we were just dropping events entirely.  

ie, I think Chad's comment is valid.  Not having idempotent behavior means we can easily get into a state where the data doesn't accurately reflect state and we can't trust the metrics.

Reply:
I don't think we can have idempotent behavior. The time between Created, Updated, and Completed is an unknown quantity potentially stretching far into the future. Someone may ask for changes on a PR and leave it around forever, disable or delete a subscription, etc.

Maybe the right thing to do is to simply track the non-matched events (in-flight events?)  This number should be small I think. We can measure the amount we don't know.

Reply:
Sounds reasonable